### PR TITLE
i#3699 ARM: Keep stack aligned in insert_{push,pop}_all_registers.

### PR DESCRIPTION
**Pull request by egrimley-arm**
dd1589c0c modified dstack_offs in insert_push_all_registers without generating an instruction that modifies SP. So add a SUB SP, SP, #4 there, and a corresponding ADD SP, SP, #4 in insert_pop_all_registers.

This made the test client.cleancallparams pass.

Issue: #3699

### CODE REVIEW

**Comment by derekbruening** [comment]
Switched reviewer for load balancing.  Some of these could presumably also go to @AssadHashmi or others.

File: core/arch/aarchxx/mangle.c [hunk 1]
```diff
@@ -665,6 +665,8 @@ insert_push_all_registers(dcontext_t *dcontext, clean_call_info_t *cci,
     dstack_offs += 15 * XSP_SZ;
 
     /* Make dstack_offs 8-byte algined, as we only accounted for 17 4-byte slots. */
```
Comment (abhinav92003) [inline core/arch/aarchxx/mangle.c:667]: nit: s/algined/aligned/

Comment (abhinav92003) [inline core/arch/aarchxx/mangle.c:667]: The accounting in this statement probably does not include the adjustment made to dstack_offs for proc_num_simd_registers at L600. Does it assume it must be a multiple of 8? Do we need an ASSERT?

File: core/arch/aarchxx/mangle.c [hunk 2]
```diff
@@ -665,6 +665,8 @@ insert_push_all_registers(dcontext_t *dcontext, clean_call_info_t *cci,
...
+    PRE(ilist, instr,
```
Comment (abhinav92003) [inline core/arch/aarchxx/mangle.c:668]: Assuming there's an odd number of XSP_SZ-sized writes to the stack and that's why we need this fix: I was trying to locate the XSP_SZ-sized stack write with the missing parity. Is it the one on L655? But that happens only for DR_ISA_ARM_THUMB isn't it?

Maybe a comment clarifying the accounting would help.

File: core/arch/aarchxx/mangle.c [hunk 3]
```diff
@@ -769,6 +771,8 @@ insert_pop_all_registers(dcontext_t *dcontext, clean_call_info_t *cci, instrlist
     }
 
 #else
+    PRE(ilist, instr,
```
Comment (abhinav92003) [inline core/arch/aarchxx/mangle.c:774]: A comment noting that this is to "undo" the alignment-sub we did before.

**Review by abhinav92003 — APPROVED** [review]

**Pull request merged**